staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -145,11 +268,11 @@ func fromUnstructuredViaJSON(u map[string]interface{}, obj interface{}) error {
 	return json.Unmarshal(data, obj)
 }
 
-func fromUnstructured(sv, dv reflect.Value) error {
+func fromUnstructured(sv, dv reflect.Value, ctx *fromUnstructuredContext) (strictErrs []error, err error) {
```

Comment:
on reflection, I think it's a simpler to understand if we leave all the return values of all the functions as they are in master and accumulate unknownField errors in `fromUnstructuredContext` (similar to how we accumulate pruned field paths in the threaded pruning opts/context) via a `fromUnstructuredContext#recordUnknownField(key string)` method.

that lets `recordUnknownField` be a no-op if we're not accumulating unknown fields

Reply:
done